src/resources/nfts/simplehash/utils.ts

```diff
@@ -33,90 +31,44 @@ const MAX_IMAGE_SCALE = 3;
-
-    const isMissingRequiredFields = !name || !collectionName || !contract_address || !token_id || !network;
-    const isPolygonAndNotAllowed =
-      polygonAllowlist && nft.chain === getPolygonNetworkObject().nfts.simplehashNetwork && !polygonAllowlist[lowercasedContractAddress];
-    const isGnosisAndNotPOAP =
-      nft.chain === getGnosisNetworkObject().nfts.simplehashNetwork && lowercasedContractAddress !== POAP_NFT_ADDRESS;
-
-    if (isMissingRequiredFields || isPolygonAndNotAllowed || isGnosisAndNotPOAP) {
-      return [];
-    }
-
-    return [
-      {
-        ...nft,
-        name: name,
-        contract_address: lowercasedContractAddress,
-        chain: network,
-        collection: { ...nft.collection, name: collectionName },
-        token_id: token_id,
-      },
-    ];
-  });
-}
-
 /**
  * Maps a `SimpleHashNFT` to a `UniqueAsset`.
  * @param nft `SimpleHashNFT`
  * @returns `UniqueAsset`
  */
-export function simpleHashNFTToUniqueAsset(nft: ValidatedSimpleHashNFT, address: string): UniqueAsset {
+export function simpleHashNFTToUniqueAsset(nft: SimpleHashNft, address: string): UniqueAsset {
   const collection = nft.collection;
   const lowercasedContractAddress = nft.contract_address?.toLowerCase();
 
   const { highResUrl: imageUrl, lowResUrl } = handleNFTImages({
     originalUrl: nft.image_url ?? nft.extra_metadata?.image_original_url,
-    previewUrl: nft.previews.image_small_url,
+    previewUrl: nft.previews?.image_small_url,
     mimeType: nft.image_properties?.mime_type,
   });
 
   const marketplace = nft.collection.marketplace_pages?.[0];
   const floorPrice = collection?.floor_prices?.find(
-    (floorPrice: SimpleHashFloorPrice) =>
-      floorPrice?.marketplace_id === SimpleHashMarketplaceId.OpenSea && floorPrice?.payment_token?.payment_token_id === 'ethereum.native'
+    floorPrice => floorPrice?.marketplace_id === 'opensea' && floorPrice?.payment_token?.payment_token_id === 'ethereum.native'
   );
 
   const isENS = lowercasedContractAddress === ENS_NFT_CONTRACT_ADDRESS;
 
-  const standard = nft.contract.type;
+  const standard = nft.contract?.type;
 
   const isPoap = nft.contract_address.toLowerCase() === POAP_NFT_ADDRESS;
 
-  const ownerEntry = nft.owners.find(o => o.owner_address === address);
+  const ownerEntry = nft.owners?.find(o => o.owner_address === address);
 
   return {
     animation_url: nft?.video_url ?? nft.audio_url ?? nft.model_url ?? nft.extra_metadata?.animation_original_url ?? undefined,
     asset_contract: {
       address: lowercasedContractAddress,
-      name: nft.contract.name || undefined,
-      schema_name: standard,
-      symbol: nft.contract.symbol || undefined,
+      name: nft.contract?.name ?? undefined,
+      schema_name: standard ?? undefined,
+      symbol: nft.contract?.symbol ?? undefined,
     },
     acquisition_date: ownerEntry?.last_acquired_date ?? undefined,
-    background: nft.background_color,
+    background: nft.background_color ?? null,
```

Comment:
small nit, but might be cleaner to use a Partial type here if lots of these are optional.